lib/delayed_resque/message_sending.rb

```diff
@@ -39,10 +39,16 @@ def method_missing(method, *args)
       end
 
       if @options[:unique]
-        if @options[:at] || @options[:in]
-          ::Resque.remove_delayed(@payload_class, stored_options)
-        else
-          ::Resque.dequeue(@payload_class, stored_options)
+        # remove_delayed uses @payload_class.queue rather than the queue
+        # value within stored_options to determine whether or not a job
+        # already exists. This can lead to issues when trying to remove
+        # duplicates in a non-default queue
+        @payload_class.with_queue(queue) do
```

Comment:
should we be wrapping 95% of the method in this block given the underlying uses of @queues in other libs?

Reply:
@sbeckeriv not sure I full understand the question. Are you suggesting we use this `with_queue(queue)` block pretty everywhere in this gem that is enqueueing/dequeueing jobs? I _think_ this spot with the `remove_delayed` and `dequeue` are the only spots where the `payload_class` would be introspected to determine what the queue is. I believe most of the other methods called explicitly specify the queue (e.g. `enqueue_at_with_queue`) and as such the queue name is passed in directly.

Reply:
well i would use it for this method for sure. given the root issue is deep in a 3rd party gem wrapping the method would prevent any other possible issues with using the .queue method. yes. we pass around the queue maybe because .queue was wrong? without a full audit of the gems we use we will never be sure if the queue is used incorrectly elsewhere.